src/general_manager/interface/databaseInterface.py

```
@@ -111,7 +111,13 @@ def __setAttrForWrite(
             if isinstance(value, GeneralManager):
                 value = value.identification["id"]
                 key = f"{key}_id"
-            setattr(instance, key, value)
+            try:
+                setattr(instance, key, value)
+            except (ValueError, TypeError) as e:
+                if isinstance(e, ValueError):
+                    raise ValueError(f"Invalid value for {key}: {value}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)